### PR TITLE
Fix weighted graph support for `igraph_community_edge_betweenness()`

### LINKED ISSUE #2229: Proper weight support for igraph_community_edge_betweenness()
`igraph_community_edge_betweenness()` implements the Girvan-Newman algorithms which builds a dendrogram by iteratively removing the highest betweenness edges and checking how the graph falls to pieces. Then the dendrogram is cut so as to maximize modularity.

When weights are passed to this function, it uses these weights as _edge lengths_ during the betweenness calculation. Therefore, high-weight edges are interpreted as representing weaker connections. During the modularity calculation, high-weight edges are interpreted as stronger connections, which is contradictory. We have discussed this before, and the stopgap measure was that the function issues a warning when it is given some weights, and a membership vector is requested. In fact, it issues two warnings, which looks a bit sloppy.

https://github.com/igraph/igraph/blob/develop/src/community/edge_betweenness.c#L506-L516

----

This is a proposal for dealing with this problem. There are actually weighted generalizations of the method in the literature.

Section 5.1 of https://doi.org/10.1016/j.physrep.2009.11.002 says:

> The method can be simply extended to the case of weighted graphs, by suitably generalizing the edge betweenness. The betweenness of a weighted edge equals the betweenness of the edge in the corresponding unweighted graph, divided by the weight of the edge [159].

Ref [159] is https://doi.org/10.1103/PhysRevE.70.056131 and says:

> The resulting algorithm is identical to that for the integer case: betweennesses are simply divided by the weight of the edge and the edge with the highest resulting score removed from the network.

----

In order to preserve the existing functionality of igraph while also interpreting weights reasonably, I propose to change `igraph_community_edge_betweenness()` to take _two_ weight-type arguments:

```c
igraph_error_t igraph_community_edge_betweenness(const igraph_t *graph,
                                      igraph_vector_int_t *result,
                                      igraph_vector_t *edge_betweenness,
                                      igraph_matrix_int_t *merges,
                                      igraph_vector_int_t *bridges,
                                      igraph_vector_t *modularity,
                                      igraph_vector_int_t *membership,
                                      igraph_bool_t directed,
                                      const igraph_vector_t *weights,
                                      const igraph_vector_t *lengths);
```

Edge _weights_ and edge _lengths_ should be specified separately, and either can be omitted. _Lengths_ are used in the betweenness calculation as usual. _Weights_ are used to divide the betweenness scores (as in the quotes above) and are also used in the modularity calculation. This solution preserves the current functionality (which is interesting for experimentation) while introducing the proper weighted generalization from the above quotes.

Does this sound reasonable to everyone? @vtraag ?

----

See also #1040

### PR DESCRIPTION
**Pull request by szhorvat**
Fixes #2229

### CODE REVIEW

File: src/community/edge_betweenness.c [hunk 1]
```diff
@@ -319,18 +319,26 @@ igraph_error_t igraph_community_eb_get_merges(const igraph_t *graph,
...
 
-/* Find the smallest active element in the vector */
-static igraph_integer_t igraph_i_vector_which_max_not_null(const igraph_vector_t *v,
-                                                   const bool *passive) {
-    igraph_integer_t which, i = 0, size = igraph_vector_size(v);
+/* Find the index i for which v[i] / w[i] is the smallest
+ * and the corresponding element is active (i.e. !passive[i]).
+ * If w is a null pointer then all w[i] are assumed to be 1.0.
+ * Assumes that at least one element is active. */
+static igraph_integer_t igraph_i_vector_which_max_not_null(
```
Comment (ntamas) [inline src/community/edge_betweenness.c:326]: Should be renamed to `igraph_i_vector_pair_find_max_ratio` or something like that as we now have two vectors and we are searching for the maximum in the ratio of them.

**Review by ntamas — CHANGES_REQUESTED** [review]

**Comment by ntamas** [comment]
LGTM now. Feel free to merge.

**Comment by szhorvat** [comment]
@ntamas I force pushed two changes. You can use the "compare" link on the GitHub interface to see what changed.

The first one renames the function, and adds a second entry in the changelog. The second rebases on `develop` and fixes conflicts.

**Comment by szhorvat** [comment]
@vtraag I merged this, but if you have concerns, we can still make changes.

**Pull request merged**